src/ir/lowering.rs

```diff
@@ -1665,8 +1885,8 @@ fn lower_binary_op(
     type_hint: Option<Ty>,
 ) -> Result<(Rvalue, Ty, Span), LoweringError> {
     let (lhs, lhs_ty, lhs_span) = if type_hint.is_none() {
-        let ty = find_expression_type(builder, lhs).unwrap_or_else(|| {
-            find_expression_type(builder, rhs).expect(
+        let ty = find_expression_type(builder, lhs)?.unwrap_or_else(|| {
+            find_expression_type(builder, rhs).unwrap().expect(
```

Comment:
is this unwrap safe to do? If that's the case please add a comment explaining that